File: HW/HW3/matrixMul.java
import java.util.Arrays;
import java.util.Scanner;

public class matrixMul {
    static int[][] MATRIX_MULT(int[][] A, int[][] B) {
        int rows_A = A.length;
        int cols_A = A[0].length;
        int rows_B = B.length;
        int cols_B = B[0].length;
        if(cols_A != rows_B) {
            throw new Error("Matrix multiplication is not defined.");
        }

        int[][] result = new int[rows_A][cols_B];
        for(int i = 0; i < rows_A; i++) {
            for(int j = 0; j < cols_B; j++) {
                result[i][j] = 0;
            }
        }

        for(int i = 0; i < rows_A; i++) {
            for(int j = 0; j < cols_B; j++) {
                int sum = 0;
                for(int k = 0; k < cols_A; k++) {
                    sum = sum + (A[i][k] * B[k][j]);
                    result[i][j] = sum;
                }
            }
        }

        return result;
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int[][] A = new int[2][3];
        for(int i = 0; i < 2; i++) {
            for(int j = 0; j < 3; j++) {
                System.out.print("Enter Integer: ");
                A[i][j] = in.nextInt();
            }
        }
        System.out.println(Arrays.deepToString(A));

        int[][] B = new int[3][2];
        for(int i = 0; i < 3; i++) {
            for(int j = 0; j < 2; j++) {
                System.out.print("Enter Integer: ");
                B[i][j] = in.nextInt();
            }
        }
        System.out.println(Arrays.deepToString(B));

        int[][] result = MATRIX_MULT(A,B);
        System.out.println(Arrays.deepToString(result));
    }
}

File: HW/HW3/matrixMul.py
def MATRIX_MULTIPLY(A, B):
    result = []
    rows_A = len(A)
    cols_A = len(A[0])
    rows_B = len(B)
    cols_B = len(B[0])

    if cols_A != rows_B:
        raise ValueError("Matrix multiplication not defined")

    for i in range(rows_A):
        row = []
        for j in range(cols_B):
            row.append(0)
        result.append(row)

    for i in range(rows_A):
        for j in range(cols_B):
            sum = 0
            for k in range(cols_A):
                sum = sum + (A[i][k] * B[k][j])
                result[i][j] = sum

    return result


A = [[1, 0],
     [2, 3],
     [4, 5]]

B = [[1, 3, 2],
     [3, 4, 8]]

result = MATRIX_MULTIPLY(A, B)
print(result)
